### PR TITLE
Assert well formed payload

### PR DESCRIPTION
**Pull request by gannan08**

### CODE REVIEW

File: lib/KmsClient.js [hunk 1]
```diff
@@ -703,6 +703,9 @@ export class KmsClient {
       const result = await httpClient.post(url, {
         agent, headers, json: config
       });
+
+      _assert(result.data, 'result.data', 'object');
+      _assert(result.data.id, 'result.data.id', 'string');
```
Comment (mattcollier) [inline lib/KmsClient.js:708]: I don't feel like the `_handleClientError` implementation in the catch block is doing to do a good job of surfacing this class of error to the user, I don't thing this is a "client error".  This is a response validation error, so maybe this validation step should happen after we exit this phase in another try/catch block.

Comment (mattcollier) [reply lib/KmsClient.js:708]: Also, is it easy to write test vectors for these error condtions?

File: lib/KmsClient.js [hunk 2]
```diff
@@ -700,19 +701,19 @@ export class KmsClient {
...
-      const result = await httpClient.post(url, {
+      result = await httpClient.post(url, {
         agent, headers, json: config
       });
-
-      _assert(result.data, 'result.data', 'object');
-      _assert(result.data.id, 'result.data.id', 'string');
-      return result.data;
     } catch(e) {
       _handleClientError({
```
Comment (mattcollier) [inline lib/KmsClient.js:708]: I presume this throws... maybe a comment.  The helper would be better named `_throwClientError`, the lack of a clear understanding makes the lack of a `return` concerning.

**Review by mattcollier — APPROVED** [review]

**Review by dlongley — APPROVED** [review]
Seems like a good improvement. I imagine we'll want to generalize this to all the calls in the future.

File: lib/KmsClient.js (hunk 2 above)
Comment (gannan08) [reply lib/KmsClient.js:708]: Created an issue.

File: lib/KmsClient.js (hunk 1 above)
Comment (gannan08) [reply lib/KmsClient.js]: It was too involved for the given fix to implement a bad server for the purpose of testing.

**Pull request merged**